### PR TITLE
Add a configuration option `kms_key_name` to dbt_project.yml for BigQ…

### LINKED ISSUE #1829: Support KMS encryption for BigQuery
### Describe the feature
Allow for KMS encryption to be specified in project configuration. Ideally this would be done in dbt_projects.yml like:
```name: fishtown_analytics

models:
  fishtown_analytics:
  kms_key_name: 'some_kms_key'
  
    events:
      kms_key_name: 'some_other_key' 
```

but may also exist in a model's {{ config() }}

### Describe alternatives you've considered
It may be possible to do this as a post-processing step outside of dbt, or in a post-hook (although I am not aware of any way to specify KMS key in sql).

### Additional context
This feature is specific to BigQuery as it currently stands but may generalize to other databases.

### Who will this benefit?
Anyone who uses BigQuery and may need to encrypt their data with cloud KMS keys (as opposed to the default encryption BigQuery already performs).

### PR DESCRIPTION
**Pull request by kconvey**
…uery projects where a Cloud KMS key can be specified and used to encrypt models. The key is specified to bigquery in the ddl OPTIONS.

ex. 
```
models:
  fishtown_analytics:
  kms_key_name: 'projects/PROJECT_ID/locations/global/keyRings/test/cryptoKeys/quickstart'
    
    my_model:
      kms_key_name: none   # disables kms encryption
```

Fixes #1829

### CODE REVIEW

**Comment by kconvey** [comment]
CLA should be signed now :)

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql [hunk 1]
```diff
@@ -36,24 +36,25 @@
   {% if temporary %}
     {% do opts.update({'expiration_timestamp': 'TIMESTAMP_ADD(CURRENT_TIMESTAMP(), INTERVAL 12 hour)'}) %}
   {% endif %}
+  {% if kms_key_name %}
+    {% do opts.update({'kms_key_name': "'" ~ kms_key_name ~ "'"}) %}
```
Comment (drewbanin) [inline plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: there is _a lot_ we can do to make these settings less onerous to manage, especially if we intend to add more of them in the future. For now though, this is a-ok with me

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql [hunk 2]
```diff
@@ -36,24 +36,25 @@
...
+  {% endif %}
 
-  {% set options -%}
-    OPTIONS({% for opt_key, opt_val in opts.items() %}
-      {{ opt_key }}={{ opt_val }}{{ "," if not loop.last }}
-    {% endfor %})
-  {%- endset %}
-  {% do return(options) %}
+  OPTIONS({% for opt_key, opt_val in opts.items() %}
```
Comment (drewbanin) [inline plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: I think we want to keep some of the logic that was overwritten here. It looks to me like you may not have pulled `dev/louisa-may-alcott` before making this PR? I think this code was _just_ merged a couple of days ago.

The `{% set %}` block is used to better manage whitespace in the compiled SQL output. The `{{ "," if not loop.last }}` construct is used to ensure that these options are comma-delimited.

**Comment by drewbanin** [comment]
Thanks for opening this PR @kconvey! I think the CLA should be in good shape now (@cla-bot check).

Have a look at my comment above and let me know if you have any questions :)

File: plugins/bigquery/dbt/include/bigquery/macros/adapters.sql (hunk 2 above)
Comment (kconvey) [reply plugins/bigquery/dbt/include/bigquery/macros/adapters.sql]: I was definitely behind. Thanks for catching!

**Comment by drewbanin** [comment]
@kconvey looks like the tests here are failing because of an issue with the `snowflake-connector-python` library. Can you try pulling the latest `dev/louisa-may-alcott` into this branch and pushing up your changes?

**Comment by beckjake** [comment]
This looks great! Thanks for your contribution @kconvey - I'll merge this now.

**Pull request merged**